enterprise/frontend/src/metabase-enterprise/auth/components/SettingsSAMLForm/SettingsSAMLForm.unit.spec.tsx

```diff
@@ -0,0 +1,115 @@
+import userEvent from "@testing-library/user-event";
+import fetchMock from "fetch-mock";
+
+import {
+  findRequests,
+  setupPropertiesEndpoints,
+  setupSettingsEndpoints,
+} from "__support__/server-mocks";
+import { renderWithProviders, screen, waitFor } from "__support__/ui";
+import { createMockGroup, createMockSettings } from "metabase-types/api/mocks";
+
+import { type SAMLFormSettings, SettingsSAMLForm } from "./SettingsSAMLForm";
+
+const GROUPS = [
+  createMockGroup(),
+  createMockGroup({ id: 2, name: "Administrators" }),
+  createMockGroup({ id: 3, name: "foo" }),
+  createMockGroup({ id: 4, name: "bar" }),
+  createMockGroup({ id: 5, name: "flamingos" }),
+];
+
+const setup = async (
+  settingValues?: Partial<SAMLFormSettings> & { "saml-enabled"?: boolean },
+) => {
+  const settings = createMockSettings(settingValues ?? {});
+  setupSettingsEndpoints([]);
+  setupPropertiesEndpoints(settings);
+
+  fetchMock.get("path:/api/permissions/group", GROUPS);
+  fetchMock.put("path:/api/saml/settings", { status: 204 });
+
+  renderWithProviders(
+    <>
+      <SettingsSAMLForm />
+    </>,
+  );
+
+  await screen.findByText("Set up SAML-based SSO");
+  await waitFor(async () => {
+    const gets = await findRequests("GET");
+    expect(gets).toHaveLength(3);
+  });
+};
+
+const fields = [
+  [/SAML Identity Provider URL/, "https://example.test"],
+  [/SAML Identity Provider Certificate/, "abc-123"],
+  [/SAML Identity Provider Issuer/, "example.test.sso"],
+] as [RegExp, string][];
+
+describe("SettingsSAMLForm", () => {
+  it("Can enable SAML via form input", async () => {
+    await setup();
+
+    for (const [label, value] of fields) {
+      // can't use forEach 🫠
+      const input = await screen.findByLabelText(label);
+      await userEvent.type(input, value);
+    }
+
+    const submitButton = await screen.findByRole("button", {
+      name: "Save and enable",
+    });
+    expect(submitButton).toBeEnabled();
+    await userEvent.click(submitButton);
+
+    await screen.findByText("Success");
+
+    await waitFor(async () => {
+      const puts = await findRequests("PUT");
+      expect(puts).toHaveLength(1);
+    });
+
+    const puts = await findRequests("PUT");
+    expect(puts).toHaveLength(1);
+    const [{ url, body }] = puts;
+    expect(url).toMatch(/api\/saml\/settings/);
+    expect(body["saml-identity-provider-uri"]).toBe(fields[0][1]);
```

Comment:
Personal opinion but I generally prefer testing against literal values rather than ones saved in a variable unless there's a strong reason to do so.

If you want to stick with this approach, could I propose using an object so it would be more readable? e.g.`fields.url.label` and `fields.url.value`

Reply:
yeah, I was thinking this while I wrote it 😂  thanks for pushing on it